mining.md

```diff
@@ -270,6 +270,31 @@ An eligible miner broadcasts the completed block to the network (via [block prop
 
 ### Block Rewards
 
+Over the entire lifetime of the protocol, 1,400,000,000 FIL (`TotalIssuance`) will be given out to miners. These funds are initially held by the network account actor, and are transferred to miners in blocks that they mine. The reward amount remains fixed for a period of 1 week (20,160 blocks, the `AdjustmentPeriod`) and is then adjusted.
+
+The equation for the current block reward is of the form:
+
+```
+Reward = IV * (Decay ^ (BlockHeight / 20160))
```

Comment:
I don't think this is right.

```
HalvingPeriodBlocks = 6 * 365 * 24 * 60 * 2 = 6307200
Decay = e^(ln(0.5) / (6307200 / 20160)) = 0.9977869135615522
IV = 1400000000 * (1 - 0.9977869135615522) = 3098321
```
For the first week BlockHeight / 20160 is 0, so the first block reward is:

```
Reward = IV * Decay^0 = IV = 3098321
```

This is too high. We run out of FIL after 452 blocks (452*3098321 > 1400000000). That's less than 4 hours.

It seems the goal is to halve the FIL in the network actor roughly every 6 years. I'm assuming this based on variable names. It would be nice if that were stated. If this is the case, we want the sum of all block rewards to equal half of the remaining network actor funds each halving period. That is:

∫f(x)_dx_ = TI*(1 - .5 ^ (_x_/HPB))
f(x) = _d/dx_(TI*(1 - .5 ^ (_x_/HPB)))
f(x) = -(TI * ln(.5) / HPB) * .5 ^ (x/HPB)

Where f(x) is the block reward and x is the block height. Summing over this formula for some multiple of the halving period gives:

```
HPB*1: 700000038
HPB*2: 1050000057
HPB*3: 1225000067
```
Which is really close to the intended values.

Holding the reward constant for a week just means replacing x with u=AP*floor(x/AP). sum(-(TI * ln(.5) / HPB) * .5 ^ (u/HPB),0,N) gives:
```
HPB*1: 700775514
HPB*2: 1051163411
HPB*3: 1225000067
```
Which is probably close enough given the other sources of error.

The original is:
```
IV * (D^(x/AP))
TI * (1 - D) * (D^(x/AP))
TI * (1 - .5^(AP/HPB)) * (.5 ^ (AP/HPB)) ^ (u/HPB)
TI * (1 - .5^(AP/HPB)) * .5 ^ (u/HPB)
```
vs.
```
-(TI * ln(.5) / HPB) * .5 ^ (u/HPB)
```
So it's only different by a constant.

Reply:
Oh yeah, forgot to copy a division over from my notes. The Reward here needs to be divided by the AdjustmentPeriod

Reply:
More generally, can we explicitly state intent here (i.e. half every sex years), and then go into how that is executed upon. It may help others reading the spec generally catch things or provide suggestions for improvements of the function (obvs locked in post-testnet but may be useful until then).

Reply:
It turns out these two formulations are approximately equal when HPB >> AP because limit(a->0, (x^a-1)/a) = ln(x). TIL

Looking at your derivations, I don't get how you derived `Decay` or got `IV = TotalIssuance*(1-Decay)/AdjustmentPeriod`. `IV=TotalIssuance*ln(2)/HalvingPeriodBlocks` seems a little simpler and intuitive to me, but they're both correct within tolerances.

I like the way you express the scale as the initial value. 
I'd prefer it if you expressed the rest of function as a power of 1/2:
```
QuantizedBlockHeight = AdjustmentPeriod*⌊BlockHeight/AdjustmentPeriod⌋
Reward = IV * .5 ^ (QuantizedBlockHeight/HalvingPeriodBlocks)
```

At the very least, `Decay` should be expressed as `.5 ^ (AdjustmentPeriod/HalvingPeriodBlocks)`.